### DISCUSSION TITLE
Raspberrymatic addon Systemvariable an Homeassistant Dashboard - Aktualisierungsintervall

### DISCUSSION THREAD

**Discussion by banane-m** [OpenCCU/OpenCCU#2124] [🇩🇪 Fragen&Antworten]
Hallo,
eine kurze Frage,
wie kann man die Abtastrate bzw. den Aktualisierungsintervall der Anzeige des übergebenen Datenpunktes im Dashboard einstellen.
(Systemvariable vom Raspberrymatic addon -> über RasberryMatic Homematic(IP) Local an das Dashboard)

Vielen Dank für eure Unterstützung.

**Comment by Baxxy13** [comment]
Welche Integration wird denn genutzt?
Die Offizielle oder die Custom?

Ich kann nur zur Custom-Integration was sagen...
Hier werden die Systemvariablen der Zentrale alle 30s gepollt. Ändern kann man den Intervall nicht.
Aber man kann mittels des Service [homematicip_local.fetch_system_variables](https://github.com/danielperna84/custom_homematic/blob/devel/README.md#homematicip_localfetch_system_variables) die Systemvariablen auch manuell pollen.

**Comment by banane-m** [comment]
Hallo Baxxy,
vielen Dank für deine schnelle Hilfe. 
Schade, dass man den Zyklus nicht einstellen kann, werde es mit den Dienst versuchen.

**Reply by Baxxy13** [reply]
Ein Pollen der Daten in sehr kurzen Abständen ist keine gute Idee, dafür ist die CCU/RM nicht gemacht.
Ich persönlich würde das Polling gerne ganz abschalten (brauche die SysVars in HA nicht), was aber auch nicht geht.

Idealerweise baut man seine Logiken Step by Step auf HA um, so das man irgendwann ganz auf die SysVars verzichten kann.

**Reply by jens-maus** [reply]
> Ein Pollen der Daten in sehr kurzen Abständen ist keine gute Idee, dafür ist die CCU/RM nicht gemacht. Ich persönlich würde das Polling gerne ganz abschalten (brauche die SysVars in HA nicht), was aber auch nicht geht.

BTW: Hab schon länger den prinzipiellen Plan ReGaHss eine zusätzliche XMLRPC Schnittstelle zu verpassen um die Systemvariablen genauso wie die Geräte aus der Ferne abfragen (siehe https://github.com/jens-maus/RaspberryMatic/issues/258). Muss man halt nur mal dazu kommen :)

**Reply by Baxxy13** [reply]
Interessant... und schon ganz schön angestaubt. :wink:

Aber braucht man das wirklich? 
Die SysVar-Änderungen müssten ja über XMLRPC "gepusht" werden, so wie die Daten der Geräte auch.
Und wenn da einer "hunderte" SysVars hat... ob das die alte Technik mitmacht?

Der schnellste Weg ohne Polling geht aktuell über den CCU-Jack, da wird alle 3s auf Änderungen geprüft und bei Bedarf dann per MQTT an den Client "gepusht".

**Comment by Baxxy13** [comment]
Zur Info für @banane-m 

Ich hatte das Thema gestern als Idee bei der Custom Integration vorgetragen.
#https://github.com/danielperna84/hahomematic/discussions/785

So wie es aussieht bekommst du deinen einstellbaren Polling-Intervall...
...natürlich nur wenn du auch die Custom Integration nutzt.

**Reply by jens-maus** [reply]
> ...natürlich nur wenn du auch die Custom Integration nutzt.

Gibt es eigentlich noch überhaupt noch einen Grund die uralte Integration? Ich denke es wird mal Zeit das die "Custom Integration" endlich direkt ins HomeAssistant Projekt wandert weil sonst zuviele falsch Abbiegen...

**Reply by Baxxy13** [reply]
Ich hatte damals die alte offizielle Integration mal kurz getestet. 
Hatte aber nicht lange gedauert bis ich festgestellt hatte das das meinen Anforderungen nicht gerecht wird.
Daraufhin hatte ich das Projekt Homeassistant erstmal ad acta gelegt.

Als dann die neue Custom-Integration auftauchte war ich ziemlich begeistert. Das Ding hatte zwar noch jede Menge Ecken und Kanten aber man konnte erkennen das das gut werden kann. 

Inzwischen schnurrt das Ding und es gibt kaum noch Fehler oder Mängel. 
Neue Features oder Bugfixes werden gerne mal von einen Tag auf den anderen implementiert, da ist der Entwickler super fix bei der Sache. 
Das wird vermutlich nicht mehr so einfach laufen wenn es eine offizielle Core-Integration wäre.

Von mir aus kann das daher immer eine Custom Integration bleiben.

**Reply by jens-maus** [reply]
> Neue Features oder Bugfixes werden gerne mal von einen Tag auf den anderen implementiert, da ist der Entwickler super fix bei der Sache. Das wird vermutlich nicht mehr so einfach laufen wenn es eine offizielle Core-Integration wäre.

Wenn die Entwickler sich direkt als HA Entwickler registrieren+integrieren geht das auch recht flott. Da kommen ja auch ständig neue Versionen raus. Und soweit ich weiss ist das zumindest der mittelfristige Plan, das die jetzige Custom Integration eine Core Integration wird. Und selbst wenn die Entwicklung dann etwas langsamer wird, hindert die entwickler ja niemand dran quasi eine nightly snapshot oder development version weiterhin als custom integration rauszubringen und die core integration quasi als LTS Version zur Verfügung zu stellen. Müsste man mal @SukramJ fragen.

**Reply by SukramJ** [reply]
Für einen möglichen Wechsel in den Core gibt es [hier](https://github.com/danielperna84/hahomematic/discussions/381) eine laufende Diskussion.
Ich tendiere aktuell dazu (noch) nicht in den Core zu wechseln. Mich stört hier die derzeit vom HA-Team geforderte Migration der Alt-Integration.
Sollten wir aber mal in den Core wechseln, würde ich zum jetzigen Zeitpunkt ausschliessen, dass noch zusätzlich eine Custom Component angeboten wird.
